.github/workflows/beam_PostCommit_Java_Examples_Dataflow_V2_Java.yml

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: PostCommit Java Examples Dataflow V2 Java
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.comment.body || github.event.sender.login}}'
+  cancel-in-progress: true
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: write
+  checks: write
+  contents: read
+  deployments: read
+  id-token: none
+  issues: write
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_PostCommit_Java_Examples_Dataflow_V2_Java:
+    name: ${{matrix.job_name}} (${{matrix.job_phrase_1}}${{matrix.job_phrase_2}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 180
+    strategy:
+      fail-fast: false
+      matrix:
+        job_name: [beam_PostCommit_Java_Examples_Dataflow_V2_Java]
+        job_phrase_1: [Run Java ]
+        job_phrase_2: [Examples on Dataflow Runner V2]
+        java_version: ['11', '17']
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      (startswith(github.event.comment.body, 'Run Java') &&
+         endswith(github.event.comment.body, 'Examples on Dataflow Runner V2'))
```

Comment:
I think we should use `contains` so we are tolerant of trailing whitespace/other chars

Reply:
@damccorm Does it make sense? We currently use strict checking to tell whether a comment is as expected or not (e.g. https://github.com/apache/beam/pull/28352/files#diff-22ba8e2fcd77c357a00841ff632eb4c174f1b3aee4863782ff150e9b27e609aaR65). If we want to handle extra whitespace/other chars I think we need to update all workflows in a separate PR.

Reply:
It's done.

Reply:
Ah you're right that we do this with other workflows, though we do use startswith for other matrix jobs...

With that in mind, I'm good with either approach here (endswith or contains), so we can just leave it as is for this one.